lib/ansible/modules/cloud/vmware/vmware_vm_sp.py

```diff
@@ -0,0 +1,211 @@
+            profile = vim.VirtualMachineDefinedProfileSpec(profileId=self.profile.profileId.uniqueId)
+            vmSpec = vim.VirtualMachineConfigSpec(vmProfile=[profile], deviceChange=device_change)
+            state, error = wait_for_task(vm.ReconfigVM_Task(spec=vmSpec))
+            if state:
+                self.module.exit_json(changed=True, msg="Attached Profile {} to VM {}".format(self.profilename, self.vmname))
+            else:
+                self.module.fail_json(changed=False, msg="Failed to attach Profile {} to VM {}. Error {}".format(self.profilename, self.vmname, error))
+
+    def reset_sp_vm(self):
+        profiles = self.spbmclient.get_profiles()
+        if not profiles:
+            self.module.fail_json(changed=False, msg="Could not retrieve Storage Profile Information from vCenter")
+        else:
+            vm = find_vm_by_name(content=self.content, vm_name=self.vmname)
+            if vm is None:
+                self.module.fail_json(changed=False, msg="Did not find VM {} in vCenter".format(self.vmname))
+
+            # Find Where VM Home is Residing
+            vmPathName = vm.config.files.vmPathName
+            vmx_datastore = vmPathName.partition(']')[0].replace('[', '')
+            vmx_profile = self.spbmclient.get_ds_default_profile(find_datastore_by_name(content=self.content, datastore_name=vmx_datastore))
+
+            disks = []
+            devices = vm.config.hardware.device
+            for device in devices:
+                if device:
+                    if isinstance(device, vim.vm.device.VirtualDisk):
+                        disks.append(device)
+            device_change = []
+            for disk in disks:
+                datastore_profile = self.spbmclient.get_ds_default_profile(disk.backing.datastore)
+                profile = vim.VirtualMachineDefinedProfileSpec(profileId=datastore_profile.uniqueId)
+                device_change.append(vim.VirtualDeviceConfigSpec(device=disk, operation=vim.VirtualDeviceConfigSpecOperation.edit, profile=[profile]))
+            profile = vim.VirtualMachineDefinedProfileSpec(profileId=vmx_profile.uniqueId)
+            vmSpec = vim.VirtualMachineConfigSpec(vmProfile=[profile], deviceChange=device_change)
+            state, error = wait_for_task(vm.ReconfigVM_Task(spec=vmSpec))
+            if state:
+                self.module.exit_json(changed=True, msg="Detached Profile {} from VM {}".format(self.profilename, self.vmname))
+            else:
+                self.module.fail_json(changed=False, msg="Failed to Detach Profile from VM {}. Error {}".format(self.vmname, error))
+
+    def check_compliance_vm(self):
+            profiles = self.spbmclient.get_profiles()
+            if not profiles:
+                self.module.fail_json(changed=False, msg="Could not retrieve Storage Profile Information from vCenter")
+            else:
+                for profile in profiles:
+                    print(profile.name)
```

Comment:
Do not use print statement. All module output is supposed to be returned as JSON.